File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-1.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        int n=s.size();
        int m=t.size();
        string match="";
        int i=0;
        int j=0;
        while(i<n && j<m){
            if(s[i]==t[j]){
                match =match + t[j];
                j++;
            }
            i++;
        }
        int k=match.size();
        return m-k;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-1.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        match={}
        for i in range(len(strs)):
            temp = strs[i]
            temp = sorted(temp)
            temp= "".join(temp)
            if temp not in match:
                match[temp]=[]
            match[temp].append(strs[i])
        ans=[]
        for key in match:
            current_group=match[key]
            ans.append(current_group)
        return ans